scripts/sparcc.py

```diff
@@ -0,0 +1,132 @@
+# python 2
+# conda install biom-format -c bioconda
+# conda install matplotlib  # pysurvey setup.py is incomplete
+# pip install -e "hg+https://bitbucket.org/yonatanf/pysurvey#egg=pysurvey"
+# note: the pysurvey build in pypi does not work. unsure why.
+
+import click
+
+# this little gem is because pysurvey depends on an older version of scipy
+# and attempts to import nanmedian from scipy.stats. This is a dirty hack.
+import scipy.stats as ss
+import scipy
+ss.nanmedian = scipy.nanmedian
+
+import pysurvey as ps
+import biom
+import pandas as pd
+
+
+def _biom_to_pysurvey_mat(table):
+    """Convert a BIOM table to a compatible pysurvey DataFrame
+    Parameters
+    ----------
+    table : biom.Table
+        The BIOM table
+    Returns
+    -------
+    DataFrame
+        A pandas DataFrame representing the BIOM table where the rows are
+        samples and the columns are observations. The sample identifiers are
+        stripped as they are not used, and any BIOM metadata are ignored
+    """
+    table = biom.load_table(table)
+    mat = table.matrix_data.toarray().T
+    return pd.DataFrame(mat, columns=table.ids(axis='observation'),
+                        index=table.ids())
+
+
+def _format_edges(fp, sparcc_z, threshold):
+    """Generate edges
+    Parameters
+    ----------
+    fp : open file
+        Where to write the results
+    sparcc_z : pd.DataFrame
+        A pandas DataFrame containing the correlations
+    threshold : float
+        A minimum absolute correlation threshold value
+    """
+    fp.write('Feature1\tFeature2\tRho\tRho_pos_neg\n')
+    for i, (idx, row) in enumerate(sparcc_z.iterrows()):
+
+        # i+1 for upper triangle
+        for col, v in zip(sparcc_z.columns[i+1:], row[i+1:]):
+            if abs(v) >= threshold:
+                fp.write('%s\t%s\t%f\t%d\n' % (idx, col, v, v > 0))
+
+
+def _format_nodes(fp, table):
+    """Format the nodes file
+    Paramters
+    ---------
+    fp : open file
+        Where to write the results too
+    table : file path
+        A file path to the BIOM table
+    """
+    header = ['Feature1', 'Fsum']
+
+    table = biom.load_table(table)
+    _, _, tmp_md = next(table.iter(axis='observation'))
+    header.extend(sorted(tmp_md.keys()))
+
+    fp.write("\t".join(header))
+    fp.write('\n')
+
+    for values, id_, md in table.iter(axis='observation'):
+        line = [str(id_), str(values.sum())]
```

Comment:
Adding a feature sum into the node table, so that users can color the nodes by abundance in Cytoscape.